pyproject.toml

```diff
@@ -0,0 +1,31 @@
+[build-system]
+# Minimum requirements for the build system to execute.
+requires = ["setuptools>=42.0.0", "wheel>=0.34.2", "Cython", "oldest-supported-numpy", "pybind11>=2.5.0", "Mako"]  # PEP 508 specifications.
+build-backend = "setuptools.build_meta"
+
+
+[tool.cibuildwheel]
+test-command = "pytest {project}/test"
+test-extras = ["test"]
+
+[tool.cibuildwheel.linux]
+test-command = ""
+skip = "*-musllinux*" # lack of opencl bindings. currently avaliable only on edge channel
+before-all = [
+    "yum install -y git openssl-devel ruby",
+    "bash {package}/scripts/build-ocl.sh",
```

Comment:
Use this for all flavours including musllinux

Reply:
I cannot. musllinux does not have yum, it uses `apk` for installing packages. The install command for muslinux is in line 21. But `opencl-dev` is present only in edge channel. So it is impossible to use it from official wheel build container. If it is important it is possible to workaround this by compile opencl related things from source.

Reply:
Yes, it's important to make sure the opencl be built from source with the patch.

Reply:
> Yes, it's important to make sure the opencl be built from source with the patch.

could you point to documentation that I should read?

Reply:
There's no documentation. We patch ocl-icd to look at ICDs inside pyopencl install dir so that `conda install pyopencl[pocl]` works

Reply:
Ok, I see patch of ocl-icd. But to build ocl-icd you need to have installed opencl-dev package. At least OpenCL headers. So maybe compile https://github.com/KhronosGroup/OpenCL-Headers may work?

Reply:
No, you don't. See https://github.com/inducer/pyopencl/blob/main/scripts/build-wheels.sh#L41-L51

Reply:
here openssl-devel fpackage is installed https://github.com/inducer/pyopencl/blob/1205024e2b9e69ee10ab27c0168ff987a8245c88/scripts/build-wheels.sh#L39

Reply:
yes, openssl-devel, not opencl-dev

Reply:
I was pretty sure thath they are same texts. stupid dislection 

test musllinux here https://github.com/Czaki/pyopencl/tree/test_macos_build

Reply:
You are right. muslllinux build steep added. I do not know why I do not notice that they are different packages.